crates/core/src/dbs/file.rs

```diff
@@ -46,12 +44,11 @@ impl FileCollector {
 	pub(super) async fn push(&mut self, value: Value) -> Result<(), Error> {
 		if let Some(mut writer) = self.writer.take() {
 			#[cfg(not(target_family = "wasm"))]
-			let writer = spawn_blocking(move || {
+			let writer = crate::exe::spawn(move || {
 				writer.push(value)?;
 				Ok::<FileWriter, Error>(writer)
 			})
-			.await
```

Comment:
It think this `spawn_blocking` is used her correctly. Tokio itself suggest rayon for computationally expensive functions that might block because of how long running they are, however here the closure just write so it is blocked by IO not CPU.